src/main/java/ch/njol/skript/effects/EffOpenInventory.java

```diff
@@ -34,125 +36,201 @@
-			invType = HOPPER;
-		} else if (parseResult.mark >= 2) {
-			openFlag = parseResult.mark ^ 2;
-			invType = ANVIL;
-		} else if (parseResult.mark >= 1) {
-			openFlag = parseResult.mark ^ 1;
-			invType = CHEST;
-		} else if (parseResult.mark >= 0) {
-			invType = WORKBENCH;
-			openFlag = parseResult.mark ^ 0;
-		} else {
-			openFlag = parseResult.mark;
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern == 1) {
+			open = true;
+			assert !parseResult.tags.isEmpty() : "Syntax incorrectly handled parse result tags.";
+			syntax = OpenableInventorySyntax.valueOf(parseResult.tags.get(0).toUpperCase(Locale.ENGLISH));
+			if (!Skript.isRunningMinecraft(syntax.getVersion())) {
+				Skript.error("Opening an inventory of type '" + syntax.toString().toLowerCase(Locale.ENGLISH) + "' is only present on Minecraft version " + syntax.getVersion());
+				return false;
+			}
+			if (!syntax.doesMethodExist()) {
+				Skript.error(syntax.getMethodError());
+				return false;
+			}
 		}
-		
-		open = matchedPattern == 0;
-		invi = open ? exprs[0] : null;
+		inventory = matchedPattern == 0 ? exprs[0] : null;
 		players = (Expression<Player>) exprs[exprs.length - 1];
-		if (openFlag == 1 && invi != null) {
-			Skript.warning("Using 'show' inventory instead of 'open' is not recommended as it will eventually show an unmodifiable view of the inventory in the future.");
+		if (inventory instanceof Literal) {
+			Literal<?> literal = (Literal<?>) inventory;
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				Skript.error("You can't open a '" + literal.toString() + "' inventory to players. It's not creatable.");
+				return false;
+			}
 		}
 		return true;
 	}
-	
+
 	@Override
-	protected void execute(final Event e) {
-		if (invi != null) {
-			Inventory i;
-			
-			assert invi != null;
-			Object o = invi.getSingle(e);
-			if (o instanceof Inventory) {
-				i = (Inventory) o;
-			} else if (o instanceof InventoryType) {
-				i = Bukkit.createInventory(null, (InventoryType) o);
-			} else {
-				return;
-			}
-			
-			if (i == null)
+	protected void execute(Event event) {
+		if (inventory != null) { // Show
+			Inventory inventory = null;
+			Object object = this.inventory.getSingle(event);
+			if (object == null)
 				return;
-			for (final Player p : players.getArray(e)) {
-				try {
-					p.openInventory(i);
-				} catch (IllegalArgumentException ex){
-					Skript.error("You can't open a " + i.getType().name().toLowerCase(Locale.ENGLISH).replaceAll("_", "") + " inventory to a player.");
+			for (Player player : players.getArray(event)) {
+				if (object instanceof Inventory) {
+					inventory = (Inventory) object;
+				} else if (object instanceof InventoryType) {
+					inventory = Bukkit.createInventory(player, (InventoryType) object);
```

Comment:
We should still do an isCreatable check here

Reply:
Do you think it should be an assert then, because it shouldn't be needed.

Reply:
Why wouldn't it be needed?

Reply:
Because it's already pre-conditioned in the init method.

Reply:
Right but that will only work for literals, which we aren't guaranteed to be dealing with

Reply:
An InventoryType is a literal

Reply:
I've changed this to a try and catch because it can throw NullPointerException due to the issues highlighted in the original issue.

Reply:
I mean if it were to be in a variable or something it wouldn't be

Reply:
The try catch is a better implementation as this can throw NullPointerException

Reply:
Tbf that's a server implementation issue and not something we should need to handle

Reply:
I would say we should include both then. The try-catch should server as a last-case save, not the primary one.